addons/website_sale/controllers/main.py

```diff
@@ -576,12 +576,30 @@ def checkout_form_validate(self, mode, all_form_values, data):
         error = dict()
         error_message = []
 
-        # prevent name change if invoices exist
         if data.get('partner_id'):
-            partner = request.env['res.partner'].browse(int(data['partner_id']))
-            if partner.exists() and partner.sudo().name and not partner.sudo().can_edit_vat() and 'name' in data and (data['name'] or False) != (partner.sudo().name or False):
+            partner_su = request.env['res.partner'].sudo().browse(int(data['partner_id'])).exists()
+            name_change = partner_su and 'name' in data and data['name'] != partner_su.name
+            email_change = partner_su and 'email' in data and data['email'] != partner_su.email
+
+            # Prevent changing the partner name if invoices have been issued.
+            if name_change and not partner_su.can_edit_vat():
                 error['name'] = 'error'
-                error_message.append(_('Changing your name is not allowed once invoices have been issued for your account. Please contact us directly for this operation.'))
+                error_message.append(_(
+                    "Changing your name is not allowed once invoices have been issued for your"
+                    " account. Please contact us directly for this operation."
+                ))
+
+            # Prevent change the partner name or email if it is an internal user.
+            if (name_change or email_change) and not all(partner_su.user_ids.mapped('share')):
```

Comment:
Maybe a `elif` here to avoid a double warning.

Reply:
Would that be a problem to display both warnings at once? (real question)

Reply:
the first blocking point of pagu was the warning, now we display more warning. while only one will be enough.
Refactoring in v14 seems wrong idea imo. You walnut the fix with this last change !

Some small changes like `or False != or False` probably there to avoid a warning in case of name not set since `False != '' is True`. 

The previous version was verbose du to duplicate condition but clear and the risk lower than now !

Reply:
More than verbose, it was really not clear, IMHO. I'm not a fan of "refactorings" in stable either but, in this case, I believe it was necessary because obscure/overly complicated code encourages producing new similar code and is way harder to maintain than plain dumb code. I also investigated all the cases covered by the previous implementation and asked @paju-odoo if two warnings were fine, which she confirmed it was, especially since it was already tested like that in the previous implem.

Reply:
Yeah I think it's better we give all the warning reason(s) to the user (in case they "try" to get around and find another warning pops out). Not very likely in this case, that an internal user switches to an external account with an existing invoice and then try to change the billing address, but we never know :rofl: